module/zcommon/zfs_uio.c

```diff
@@ -79,8 +81,19 @@ uiomove_iov(void *p, size_t n, enum uio_rw rw, struct uio *uio)
 				if (copy_to_user(iov->iov_base+skip, p, cnt))
 					return (EFAULT);
 			} else {
-				if (copy_from_user(p, iov->iov_base+skip, cnt))
-					return (EFAULT);
+				if (fault_disable) {
```

Comment:
Perhaps we could do this with a flag in the `uio_t` instead to avoid changing all these interfaces.